gdnative-core/src/init.rs

```diff
@@ -0,0 +1,46 @@
+#[allow(unused)]
+pub struct InitCallbackInfo {
+    pub in_editor: bool,
+    active_library_path: String,
+    options: *mut crate::sys::godot_gdnative_init_options,
+}
+
+#[allow(dead_code)]
+impl InitCallbackInfo {
+    pub unsafe fn new(options: *mut crate::sys::godot_gdnative_init_options) -> Self {
+        assert!(!options.is_null(), "InitCallbackOptions were NULL");
+        let crate::sys::godot_gdnative_init_options {
+            in_editor,
+            active_library_path,
+            ..
+        } = *options;
+
+        let active_library_path = crate::core_types::GodotString::from_sys(*active_library_path);
+        let active_library_path = active_library_path.to_string();
+
+        Self {
+            options,
+            in_editor,
+            active_library_path,
+        }
+    }
+
+    pub fn report_loading_error(&self, message: &'static str) {
+        let crate::sys::godot_gdnative_init_options {
+            report_loading_error,
+            gd_native_library,
+            ..
+        } = unsafe { *self.options };
+
+        if let Some(report_loading_error_fn) = report_loading_error {
+            let message =
+                std::ffi::CString::new(message.as_bytes()).expect("message should not have a NULL");
+            let library = gd_native_library;
+            let message_ptr = message.as_c_str().as_ptr();
+            unsafe {
+                report_loading_error_fn(library, message_ptr);
```

Comment:
This will occasionally end in a memory error:
`godot.osx.opt.tools.64.editor(94936,0x114799dc0) malloc: Incorrect checksum for freed object 0x7fea9126a4b8: probably modified after being freed.`

The pointer that is referenced is neither the message or library.